lua/lspconfig/server_configurations/gleam.lua

```diff
@@ -0,0 +1,24 @@
+local util = require 'lspconfig.util'
+
+return {
+  default_config = {
+    cmd = {'gleam', 'lsp'},
+    filetypes = { 'gleam' },
+    root_dir = function(fname)
+      return util.root_pattern('gleam.toml', '.git')(fname) or vim.loop.os_homedir()
```

Comment:
remove the homedir fallback, and the pattern should reduce to `root_dir = util.root_pattern('gleam.tom', '.git')`